src/pybind/mgr/telemetry/module.py

```diff
@@ -673,16 +673,23 @@ def send(self, report, endpoint=None):
         for e in endpoint:
             if e == 'ceph':
                 self.log.info('Sending ceph report to: %s', self.url)
-                resp = requests.put(url=self.url, json=report, proxies=proxies)
-                if not resp.ok:
-                    self.log.error("Report send failed: %d %s %s" %
-                                   (resp.status_code, resp.reason, resp.text))
-                    failed.append('Failed to send report to %s: %d %s %s' % (
-                        self.url,
-                        resp.status_code,
-                        resp.reason,
-                        resp.text
-                    ))
+                fail_reason = None
+                try:
+                    resp = requests.put(url=self.url, json=report,
+                                        proxies=proxies)
+                    if not resp.ok:
```

Comment:
should probably change to "raise_for_status()" and then handle everything as an exception; avoids duplicating code